cli/flox-rust-sdk/src/providers/services.rs

```diff
@@ -405,4 +581,202 @@ mod tests {
+    /// and maintains the order of logs from each process.
+    #[test]
+    fn test_multiple_process_logs_received_in_order() {
+        let instance = TestProcessComposeInstance::start(&ProcessComposeConfig {
+            processes: [
+                ("foo".to_string(), ProcessConfig {
+                    command: "i=0; while true; do i=$((i+1)); echo \"$((i))\"; sleep 0.1; done"
+                        .to_string(),
+                    vars: None,
+                }),
+                ("bar".to_string(), ProcessConfig {
+                    command: "i=0; while true; do i=$((i+1)); echo \"$((i))\"; sleep 0.1; done"
+                        .to_string(),
+                    vars: None,
+                }),
+            ]
+            .into(),
+        });
+
+        let stream = ProcessComposeLogStream::new(instance.socket(), ["foo", "bar"])
+            .unwrap()
+            .map(|line| line.unwrap())
+            .take(10);
+
+        let groups = stream.group_by(|line| line.process.clone());
+        let groups = groups.into_iter().collect::<HashMap<_, _>>();
+
+        assert_eq!(groups.len(), 2, "expected two processes");
+
+        for (process, lines) in groups.into_iter() {
+            let lines = lines.collect::<Vec<_>>();
+            let lines_sorted = lines
+                .clone()
+                .into_iter()
+                .sorted_by_key(|line| line.message.clone())
+                .collect::<Vec<_>>();
+            assert_eq!(
+                lines, lines_sorted,
+                "{process} lines out of order: {lines:#?}"
+            );
+        }
+    }
+
+    /// Test that [ProcessComposeLogStream] returns an error when the socket doesn't exist.
+    #[test]
+    fn test_socket_gone() {
+        let instance = TestProcessComposeInstance::start(&ProcessComposeConfig {
+            processes: [("foo".to_string(), ProcessConfig {
+                command: String::from(
+                    "i=0; while true; do i=$((i+1)); echo foo \"$((i))\"; sleep 0.1; done",
+                ),
+                vars: None,
+            })]
+            .into(),
+        });
+
+        let socket = instance.socket().to_path_buf();
+        instance.stop();
+
+        let mut stream = ProcessComposeLogStream::new(socket, ["foo"]).unwrap();
+
+        let first_message = stream.next().unwrap();
+        // the only error in the stream should be that the socket doesn't exist
+        assert!(
+            matches!(
+                first_message,
+                Err(ServiceError::LoggedError(LoggedError::SocketDoesntExist))
+            ),
+            "expected socket error, got {:?}",
+            first_message
+        );
+
+        let remaining_messages = stream.collect::<Vec<_>>();
+        assert!(
+            remaining_messages.is_empty(),
+            "expected no more messages, got: {:?}",
+            remaining_messages
+        );
+    }
```

Comment:
praise: tests look good! I like that we're testing actual integration with `process-compose`, but not in a way that's super heavyweight